spec/controllers/spree/checkout_controller_spec.rb

```diff
@@ -25,12 +25,32 @@
         end
       end
 
-      context 'when authenticated as guest' do
+      context 'when not authenticated as guest' do
         it 'redirects to registration step' do
           spree_get :edit, { state: 'address' }
           expect(response).to redirect_to spree.checkout_registration_path
         end
       end
+
+      context 'when authenticated as guest' do
+        before { order.email = 'guest@solidus.io' }
+
+        it 'proceeds to the first checkout step' do
+          spree_get :edit, { state: 'address' }
+          expect(response).to render_template :edit
+        end
+
+        context 'when guest checkout not allowed' do
+          before do
+            Spree::Config.set(allow_guest_checkout: false)
```

Comment:
This conflicts with the feature specs, which expect guest checkout to be allowed. Since the default setting is `true`, it should be reset after this spec group has run.